dateparser/date.py

```diff
@@ -136,6 +136,9 @@ def parse_with_formats(date_string, date_formats, settings):
     :returns: :class:`datetime.datetime`, dict or None
 
     """
+    if isinstance(date_formats, six.string_types):
+        warn(_DateLocaleParser.DATE_FORMATS_ERROR_MESSAGE, FutureWarning)
```

Comment:
I'm not sure about the warning part.
Since we already support strings as formats, why should we throw the `Date formats should be list, tuple or set of strings` warning?

Reply:
I don’t really mind either way, I just went with a deprecation warning because it seemed consistent with the current code (due to https://github.com/scrapinghub/dateparser/pull/56/commits/136d301e250540b3fa76bf78a33f345e81fc9e6a).

If we don’t issue a deprecation warning here, maybe we should also remove the warnings from other places. What do you think?

Reply:
Good digging!
Well, let's leave the warning then.